hilti/toolchain/src/compiler/visitors/operator-resolver.cc

```diff
@@ -152,6 +154,36 @@ struct Normalizer : public hilti::visitor::PostOrder<void, Normalizer> {
             }
         }
     }
+
+    void operator()(const expression::Assign& assign, position_t p) {
+        // Rewrite assignments to map elements to use the `index_assign` operator.
+        auto& lhs = assign.childs().front();
+        if ( const auto& index_non_const = lhs.tryAs<operator_::map::IndexNonConst>() ) {
+            const auto map = index_non_const->op0();
+
+            const auto map_type = map.type().as<type::Map>();
+            const auto key_type = map_type.keyType();
+            auto value_type = map_type.elementType();
+
+            auto key = index_non_const->op1();
+            if ( key.type() != key_type ) {
```

Comment:
Actually there's a better way than what I had said earlier: if you look into `c.nexpr` instead of `c.coerced`, you can skip the `if` check if the types match (because `coerceExpression` checks that already; I hadn't thought about that).

Reply:
Ah, that's convenient.